File: composeApp/src/iosMain/kotlin/com/triapp/utils/IOSNotificationManager.kt
package com.triapp.utils

import kotlinx.coroutines.suspendCancellableCoroutine
import platform.UserNotifications.*

class IOSNotificationManager : NotificationManager {

    override suspend fun requestPermission(): Boolean {
        val center = UNUserNotificationCenter.currentNotificationCenter()
        return suspendCancellableCoroutine { continuation ->
            center.requestAuthorizationWithOptions(
                options = UNAuthorizationOptionAlert or UNAuthorizationOptionSound or UNAuthorizationOptionBadge
            ) { granted, _ ->
                continuation.resume(granted) {}
            }
        }
    }

    override fun showNotification(title: String, message: String) {
        val content = UNMutableNotificationContent().apply {
            this.title = title
            this.body = message
        }

        val request = UNNotificationRequest.requestWithIdentifier(
            identifier = "local_notification",
            content = content,
            trigger = null
        )

        UNUserNotificationCenter.currentNotificationCenter()
            .addNotificationRequest(request, withCompletionHandler = null)
    }
}

File: composeApp/src/iosMain/kotlin/com/triapp/utils/IosNotificationManager.swift
import Foundation
import UserNotifications
import UIKit

@objc class IosNotificationManager: NSObject {
    private let notificationCenter = UNUserNotificationCenter.current()

    @objc func requestPermission(completion: @escaping (Bool) -> Void) {
        notificationCenter.requestAuthorization(options: [.alert, .sound, .badge]) { granted, error in
            DispatchQueue.main.async {
                completion(granted)
            }
        }
    }

    @objc func showNotificationWithTitle(_ title: String, message: String) {
        let content = UNMutableNotificationContent()
        content.title = title
        content.body = message
        content.sound = .default

        let trigger = UNTimeIntervalNotificationTrigger(timeInterval: 1, repeats: false)
        let request = UNNotificationRequest(identifier: UUID().uuidString, content: content, trigger: trigger)

        notificationCenter.add(request) { error in
            if let error = error {
                print("Erro ao mostrar notificação: \(error)")
            }
        }
    }
}
